tools/ARIAtools/extractProduct.py

```diff
@@ -267,20 +267,48 @@ def merged_productbbox(product_dict, workdir='./', bbox_file=None, croptounion=F
     if not os.path.exists(workdir):
         os.mkdir(workdir)
 
+    # Load bbox_file if given
+    if bbox_file:
+        user_bbox=open_shapefile(bbox_file, 0, 0)
+
     # Extract/merge productBoundingBox layers
+    validation_array=[]
     for scene in product_dict:
         # Get pair name, expected in dictionary
         pair_name=scene["pair_name"][0]
         outname=os.path.join(workdir, pair_name+'.shp')
 
         # Create union of productBoundingBox layers
+        frames_in_scene=0 # keep running tally of number of valid frames
         for frame in scene["productBoundingBox"]:
             prods_bbox=open_shapefile(frame, 'productBoundingBox', 1)
+
+            # Check if frames fall within bounds of bbox file
+            if bbox_file:
+                # find area of intersection between user bbox and frame extent
+                prod_in_user_bbox=prods_bbox.intersection(user_bbox)
+
+                # report percent of frame within user bbox
+                per_in_user_bbox=prod_in_user_bbox.area/prods_bbox.area*100
+
+                # remove frame if area less than some limit
+                if per_in_user_bbox>min_percent_frame_overlap:
+                    frames_in_scene+=1
+
             if os.path.exists(outname):
                 union_bbox=open_shapefile(outname, 0, 0)
```

Comment:
@rzinke @sssangha checking the percentage overlap here.

Reply:
Hi @dbekaert and @sssangha,
Thanks for reviewing the changes. Yes, I remember the original plan was to check the overlap percentage further down in the script, possibly in section below "if bbox_file is not None: ...", but the issue is that sometimes a single frame creeps into the bbox area and a new scene is created because of it. Here is an example of the issue: 

![Screen Shot 2019-10-04 at 2 14 02 PM](https://user-images.githubusercontent.com/48299159/66447038-c4416e00-ea01-11e9-98d0-82245225be54.png)

It seemed like it was easier to throw this one frame out rather than make an entire scene, then delete the entire scene. I also would have to reverse the logic... As implemented above, the script says "the unwanted frame is most outside the user bbox, so throw it out." In the suggested case, I think it would have to say "the unwanted scene does not fill up the user bbox, so throw it out." Is there a more elegant way of doing it? What do you think?